tests/surrogate_test.py

```diff
@@ -0,0 +1,206 @@
+import numpy
+import pytest
+
+from smolyay.basis import ChebyshevFirstKind
+from smolyay.grid import SmolyakGridGenerator, IndexGrid
+from smolyay.surrogate import Surrogate
+
+
+def function_0(x):
+    """Test function 0."""
+    x1, x2, x3 = x
+    return 2*x1 + x2 - x3
+
+
+def function_0_shifted(x):
+    """Test fucntion 0 which is shifted."""
+    x1, x2, x3 = x
+    return 2*(x1 - 1) + (x2 + 1) - x3
+
+
+def function_1(x):
+    """Test fucntion 1."""
+    x1, x2 = x
+    return x1 + (2*x2**2 - 1)
+
+
+def function_2(x):
+    """Test funciton 2."""
+    return x**2 - 3 * (2 + x) - x
+
+
+def branin(x):
+    """Branin function."""
+    x1, x2 = x
+    branin1 = (x2 - 5.1 * x1 ** (2)/(4 * numpy.pi ** 2)
+               + 5 * x1 / (numpy.pi) - 6) ** 2
+    branin2 = 10 * (1 - 1 / (8 * numpy.pi)) * numpy.cos(x1)
+    branin3 = 10
+    branin_function = branin1 + branin2 + branin3
+    return branin_function
+
+
+def branin_at_grids():
+    """Evaluate Branin function at grid points."""
+    points = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(4)
+                                  )(2).points
+    transformed_points = numpy.zeros((len(points), 2))
+    transformed_points[:, 0] = numpy.polynomial.polyutils.mapdomain(
+                            numpy.array(points)[:, 0],
+                            (-1, 1), (-5, 10))
+    transformed_points[:, 1] = numpy.polynomial.polyutils.mapdomain(
+                            numpy.array(points)[:, 1],
+                            (-1, 1), (0, 15))
+    branin_at_grids = []
+    for i in transformed_points:
+        branin_at_grids.append(branin(i))
+    return branin_at_grids
+
+
+def test_initialization_1d():
+    """Test if class is properly intiallized."""
+    grid_generator = SmolyakGridGenerator(
+        ChebyshevFirstKind.make_nested_set(2))
+    domain = (-1, 1)
+    surrogate_object = Surrogate(domain, grid_generator)
+    assert numpy.allclose(surrogate_object.domain,
+                          numpy.array(domain, ndmin=2))
+    assert surrogate_object._coefficients is None
+    assert surrogate_object._grid is None
+    grid_object = surrogate_object.grid
+    assert isinstance(grid_object, IndexGrid)
+    assert surrogate_object.dimension == 1
```

Comment:
Test as it should be used:
```suggestion
    assert numpy.allclose(surrogate_object.domain, domain)
    assert surrogate_object.coefficients is None
    assert isinstance(surrogate_object.grid, IndexGrid)
    assert surrogate_object.dimension == 1
```
Should you also check `points` here?